server/src/main/java/org/opensearch/action/search/TransportSearchAction.java

```diff
@@ -1127,15 +1192,35 @@ AbstractSearchAsyncAction<? extends SearchPhaseResult> asyncSearchAction(
             ActionListener<SearchResponse> listener,
             boolean preFilter,
             ThreadPool threadPool,
-            SearchResponse.Clusters clusters
+            SearchResponse.Clusters clusters,
+            SearchRequestOperationsListener searchRequestOperationsListener
         );
     }
 
-    private List<SearchRequestOperationsListener> createSearchListenerList() {
+    private List<SearchRequestOperationsListener> createSearchListenerList(SearchRequest searchRequest, SearchTimeProvider timeProvider) {
         final List<SearchRequestOperationsListener> searchListenersList = new ArrayList<>();
+
         if (isRequestStatsEnabled) {
             searchListenersList.add(searchRequestStats);
         }
+
+        // phase_took is enabled with request param and/or cluster setting
+        // check cluster setting only when request param is unspecified
+        switch (searchRequest.isPhaseTook()) {
+            case TRUE:
+                timeProvider.setPhaseTook(true);
+                searchListenersList.add(timeProvider);
+                break;
+            case FALSE:
+                break;
+            case UNSET:
+                if (clusterService.getClusterSettings().get(TransportSearchAction.SEARCH_PHASE_TOOK_ENABLED)) {
+                    timeProvider.setPhaseTook(true);
+                    searchListenersList.add(timeProvider);
+                }
```

Comment:
Why would we ever want to enable this via a cluster parameter? That would increase the size of every search response.

Reply:
We view the `phase_took` use case similar to the `took` field which is present in all search responses. I don't believe it adds much to the response size and users are opting-in if they enable.

Reply:
I would argue that `took` also shouldn't be in search responses unless explicitly requested either. 😁 

A production application calling OpenSearch generally doesn't care how long the request took on the OpenSearch cluster. (If we want metrics on that, they should be emitted by the OpenSearch cluster.) The application just wants search results.

I would lean toward including less data by default in responses. If a human wants to trace things they can ask, but the vast majority of responses should be processed by machines. We should optimize for the 99% use-case.

Reply:
@msfroh Cluster settings is turned off by default and is up to users to enable it. This setting gives capability for users who are looking for this info as part of every response and okay for any trade-off that may exist.

Reply:
What about the trade-off of having *yet another* cluster setting that needs to be kept around on an ongoing basis for backward compatibility? Every setting added to the codebase adds an incremental (small, but non-zero) maintenance cost.

We don't have a cluster setting to enable pretty-printing by default -- we ask users to specify `?pretty` in their search request. Users who rely on the old behavior of having `total_hits` returned as an `int` instead of an object still need to specify `rest_total_hits_as_int=true` on every request. There's no cluster setting for `track_total_hits`. These are all cases where some hypothetical users may want behavior other than the default, but we ask them to do it in the search request.

Reply:
Users do track took_time as part of response to get a sense of total time taken at coordinator level as otherwise there is no metrics available(apart from shard level latencies). Though we exposed it as part of node stats lately.

Regarding ```?pretty```, I guess this is more from a command line perspective. Agree on ```rest_total_hits_as_int```. Though there are params like ```allow_partial_search_results``` which are also provided as part of cluster settings but don't want to compare it with this as such.

I am okay with just having param as well as far as there is a capability for users to get that info. Cluster setting mostly helps in cases where users want to explicitly track this at their application side and don't want to make change(appending new param) in their code as such.

Reply:
@msfroh 
We have had a number of cases where the users actively track these for their workloads to monitor latencies as p90+ . The feature will be off by default and we can consider turning it on for OpenSearch 3.0. 

As a counter point, lot of clients don't actively use the `_shards` , but we still have it in the response payload and there isn't any way to turn it off.  There are settings like `cancel_after_time_interval` which also has a cluster setting to define a default option, should the client already have capability to handle it. The maintenance cost is a one time cost IMHO.  We should be able to handle optional parameters with cluster-side defaults if we think they can be used extensively.

Reply:
> The feature will be off by default and we can consider turning it on for OpenSearch 3.0.

I sincerely hope that by the time we launch OpenSearch 3.0, we have proper logging of metrics (via the tracing framework), such that folks will be able to track these values without needing them in their search responses.

Reply:
I still disagree with the cluster setting, but I agree that it's not a huge mistake. As long as it's off by default, we're not going to unnecessarily waste network traffic.

Reply:
Yes. It is a 2-way door. Going off by default while turning on for select cases will help us assess the impact of the additional bytes. We wouldn't be having this if the existing node stat provided us a way to track percentiles, but that is not possible with a counter.